nomenklatura/store/sql.py

```diff
@@ -0,0 +1,167 @@
+from typing import Any, Generator, List, Optional, Set, Tuple
+
+from followthemoney.property import Property
+from sqlalchemy import Table, create_engine, delete, select
+from sqlalchemy.sql.selectable import Select
+
+from nomenklatura.dataset import DS
+from nomenklatura.db import (
+    DB_URL,
+    POOL_SIZE,
+    get_metadata,
+    get_statement_table,
+    get_upsert_func,
+)
+from nomenklatura.entity import CE
+from nomenklatura.resolver import Resolver
+from nomenklatura.statement import Statement
+from nomenklatura.statement.serialize import pack_sql_statement
+from nomenklatura.store import Store, View, Writer
+
+
+class SqlStore(Store[DS, CE]):
+    def __init__(
+        self,
+        dataset: DS,
+        resolver: Resolver[CE],
+        uri: str = DB_URL,
+        **engine_kwargs: Any,
+    ):
+        super().__init__(dataset, resolver)
+        engine_kwargs["pool_size"] = engine_kwargs.pop("pool_size", POOL_SIZE)
+        self.metadata = get_metadata()
+        self.engine = create_engine(uri, **engine_kwargs)
+        self.table = get_statement_table()
+        self.metadata.create_all(self.engine, tables=[self.table], checkfirst=True)
+
+    def writer(self) -> Writer[DS, CE]:
+        return SqlWriter(self)
+
+    def view(self, scope: DS, external: bool = False) -> View[DS, CE]:
+        return SqlView(self, scope, external=external)
+
+    def _iterate_stmts(self, q: Select) -> Generator[Statement, None, None]:
+        with self.engine.connect() as conn:
+            conn = conn.execution_options(stream_results=True)
```

Comment:
This has quite a bit of overhead. I wonder if we should pass an option into this func that can disable it for `get_entity`.

Reply:
https://github.com/opensanctions/nomenklatura/pull/125/commits/8ed05449e6a334a0959f520102d98954dd142d4c